src/hypothesis/core.py

```diff
@@ -503,13 +503,45 @@ class StateForActualGivenExecution(object):
     def __init__(self, test_runner, search_strategy, test, settings, random):
         self.test_runner = test_runner
         self.search_strategy = search_strategy
-        self.test = test
         self.settings = settings
         self.at_least_one_success = False
         self.last_exception = None
         self.repr_for_last_exception = None
         self.falsifying_example = None
         self.random = random
+        self.__warned_deadline = False
+        if self.settings.deadline is None:
+            self.test = test
+        else:
+            @proxies(test)
+            def timed_test(*args, **kwargs):
+                start = time.time()
+                result = test(*args, **kwargs)
+                runtime = (time.time() - start) * 1000
+                if self.settings.deadline is not_set:
+                    if (
+                        not self.__warned_deadline and
+                        runtime >= 200
+                    ):
+                        self.__warned_deadline = True
+                        note_deprecation((
+                            'Test took %.2fms to run. In future the default '
+                            'deadline setting will be 200ms, which will '
+                            'make this an error. You can set deadline to '
+                            'an explicit value of e.g. %d to turn tests '
+                            'slower than this into an error, or you can set '
+                            'it to None to disable this check entirely.') % (
+                                runtime, ceil(runtime / 100) * 100,
+                        ))
+                elif runtime >= self.settings.deadline:
+                    raise DeadlineExceeded((
+                        'Test took %.2fms, which exceeds the deadline of '
+                        '%.2fms') % (
+                            runtime, self.settings.deadline,
+                    )
+                    )
```

Comment:
Nit: these parentheses are all over the place. I think

```python
                    raise DeadlineExceeded(
                        'Test took %.2fms, which exceeds the deadline of '
                        '%.2fms' % (runtime, self.settings.deadline)
                    )
```

works just as well, no?